sound/soc/sof/ipc4-topology.c

```diff
@@ -3000,6 +3070,16 @@ static int sof_ipc4_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget
 		dev_dbg(sdev->dev, "Create widget %s (pipe %d) - ID %d, instance %d, core %d\n",
 			swidget->widget->name, swidget->pipeline_id, module_id,
 			swidget->instance_id, swidget->core);
+
+		ret = sof_ipc4_widget_setup_msg_payload(sdev, swidget, msg, ipc_data, ipc_size,
+						&ext_data, &new_size);
```

Comment:
Why don't we call this from sof_ipc4_widget_setup_msg() to avoid runtime penalty and do this one once at probe time?

Reply:
Hmm,  but do we not have the module specific payload available there, do we? That is why the payload is added only at sof_ipc4_widget_setup() phase, isn't it? We could of course pre-encode the ext_init payload, store it somewhere and then concatenate the payloads in sof_ipc4_widget_setup(), would that be better?